modules/internal/localeModels/knl/LocaleModel.chpl

```diff
@@ -223,12 +223,16 @@ module LocaleModel {
     }
 
     override proc writeThis(f) {
-      parent.writeThis(f);
+      parent!.writeThis(f);
       f <~> '.'+mlName;
     }
 
     override proc getChildCount(): int { return 0; }
-    override proc getChild(idx:int) : locale { return nil; }
+
+    override proc getChild(idx:int) : locale {
+      halt("requesting a child from a MemoryLocale locale");
+      return new locale(); //dummy
```

Comment:
Not sure why this is needed. Also, shouldn't all of these `getChild` methods return a `borrowed local` ? I can't remember if `locale` is exempt from the undecorated-is-generic-management rule right now.